### ISSUE TITLE
F35 and later builds don't recognize python3-nose2

### ISSUE THREAD

**Issue by brancomat** [ARPA-SIMC/dballe#268]
see: https://download.copr.fedorainfracloud.org/results/simc/stable/fedora-35-x86_64/03415740-dballe/builder-live.log.gz

During the build it installs `python3-nose2-0.9.1-7.fc35`

but then:
```
checking for nose2-3.1... no
checking for nose2-3... no
configure: error: Python nose not found
```

in f34 (that has `python3-nose2-0.9.1-5.fc34`) it seems to check for a different version:
```
checking python3 module: numpy... yes
checking for nose2-3.9... nose2-3.9
```

**Comment by edigiacomo** [comment]
Supporting `nose2-3.10` was not enough, because `automake` considers Python 3.10 to be 3.1:

```
checking for python3 version... 3.1
````

The bug is fixed in automake 1.16.3 (https://savannah.gnu.org/forum/forum.php?forum_id=9862) but Fedora 35 ships with version 1.16.2.

**Comment by spanezz** [comment]
Would it be the right time to consider building with meson instead of autotools?

**Comment by edigiacomo** [comment]
Interesting, in F35 there's a patch that should fix this problem: https://src.fedoraproject.org/rpms/automake/blob/f35/f/automake-1.16.2-python-version-multiple-digits.patch. See also bug https://bugzilla.redhat.com/show_bug.cgi?id=1889732

**Comment by edigiacomo** [comment]
The Fedora patch was installed, but there was a `m4/python.m4` in the upstream :)

> Would it be the right time to consider building with meson instead of autotools?

The question mark was a typo, right? Joking apart, we should really consider moving to meson.

**Comment by spanezz** [comment]
Thanks to a tip I got from friends, in 4e831f75fa936aac07154f9c478916813105d6dd I committed a way not to require nose for running tests at all. More details are in the commit message.

I like it and I'd make it official: whether we switch to meson or not, it turns out that nose is a dependency that we do not need.

In 666c33fe3af7f68878e059d3825e134874cb765b I removed nose from dependencies, let's see how it goes.
